render/src/umi/mod.rs

```diff
@@ -131,101 +162,139 @@ impl Umi {
             "BulletList" => {
                 let mut bullet_list = BulletList {
                     entries: vec![],
-                    start: serde_json::from_str(attributes.get("start").unwrap()).unwrap(),
-                    end: serde_json::from_str(attributes.get("end").unwrap()).unwrap(),
+                    start: Position::new(1, 1),
+                    end: Position::new(1, 1),
                 };
 
                 let bullet_list_depth = current_line.depth;
                 let mut current_line_index = line + 1;
-                current_line = self.fetch_next_line(current_line_index).unwrap();
+                current_line = self.fetch_next_line(current_line_index).unwrap_or_default();
 
                 while current_line.depth > bullet_list_depth {
                     if current_line.depth == bullet_list_depth + 1 {
                         // Append Element to Bullet List
                         let block = self.read_row(current_line_index);
                         let bullet_list_entry = match block {
-                            Block::BulletListEntry(block) => block,
-                            _ => panic!(),
+                            Ok(Block::BulletListEntry(block)) => block,
+                            _ => break,
                         };
                         bullet_list.entries.append(&mut vec![bullet_list_entry]);
-                    } else {
-                        break;
                     }
 
                     current_line_index += 1;
                     let fetched = self.fetch_next_line(current_line_index);
                     if fetched.is_none() {
                         break;
                     }
-                    current_line = fetched.unwrap();
+                    current_line = fetched.unwrap_or_default();
                 }
 
-                Block::BulletList(bullet_list)
+                Ok(Block::BulletList(bullet_list))
             }
             "BulletListEntry" => {
                 let mut bullet_list_entry = BulletListEntry {
                     keyword: TokenKind::from(SymbolKind::from(
-                        attributes.get("keyword").unwrap().as_str(),
+                        properties
+                            .get("keyword")
+                            .ok_or(ParserError::MissingProperty((
+                                "keyword".into(),
+                                current_line.position,
+                            )))?
+                            .as_str(),
                     ))
                     .try_into()
-                    .unwrap(),
-                    heading: self.read_inlines(attributes.get("heading").unwrap().to_string()),
+                    .ok()
+                    .ok_or(ParserError::InvalidPropertyValue((
+                        "keyword".into(),
+                        current_line.position,
+                    )))?,
+                    heading: self.read_inlines(
+                        properties
```

Comment:
the heading content should be in the `content-<lang>` column.
I missed this in the last review. sorry.

e.g.:

```
- entry heading

  Some entry body.
```

Should be
```
| kind            | depth | content-<lang>   |
| BulletList      | 0     |                  |
| BulletListEntry | 1     | entry heading    |
| Paragraph       | 2     | Some entry body. |
```